lib/core/Cas.ts

```diff
@@ -12,7 +14,18 @@ export default class Cas implements ICas {
 
   private fetch = nodeFetch;
 
-  public constructor (public uri: string) { }
+  private versionUri: string // e.g. https://127.0.0.1/version
```

Comment:
Consider hidding `versionUri` in `tryGetServiceVersion()`

Reply:
Refactoring this code out to the new ServiceVersionFetcher.ts